shardingsphere-infra/shardingsphere-infra-executor/src/main/java/org/apache/shardingsphere/infra/executor/sql/federate/execute/FederateJDBCExecutor.java

```diff
@@ -82,7 +82,7 @@ public ResultSet getResultSet() throws SQLException {
     }
     
     private ResultSet execute(final String sql, final List<Object> parameters) throws SQLException {
-        PreparedStatement statement = getConnection().prepareStatement(sql);
+        PreparedStatement statement = getConnection().prepareStatement(sql.replace(";", ""));
```

Comment:
What if the SQL contains a string contains `;`?

Reply:
> contains

@TeslaCN Yes, it may delete the wrong semicolon.